src/lib/y2partitioner/widgets/rescan_devices_button.rb

```diff
@@ -0,0 +1,56 @@
+require "yast"
+require "cwm/widget"
+require "y2partitioner/device_graphs"
+require "y2storage/storage_manager"
+
+Yast.import "Popup"
+
+module Y2Partitioner
+  module Widgets
+    # Button for rescanning system devices
+    class RescanDevicesButton < CWM::PushButton
+      def initialize
+        textdomain "storage"
+      end
+
+      def label
+        _("Rescan Devices")
+      end
+
+      # Shows a confirm message before reprobing
+      def handle
+        return nil unless continue?
+
+        reprobe
+        :redraw
+      end
+
+    private
+
+      def continue?
+        Yast::Popup.YesNo(
+          # TRANSLATORS
+          format(
+            _("Current changes will be discarted, including changes done by the proposal. " \
```

Comment:
typo: discarted -> discarded.

Anyway, I would be more explicit about the effect.... I will make a suggestion tomorrow.